Datastore/src/Operation.php

```diff
@@ -264,6 +265,14 @@ public function entity($key = null, array $entity = [], array $options = [])
      */
     public function beginTransaction($transactionOptions, array $options = [])
     {
+        // Read Only option might not be present or empty
+        // Parse only when Read Time is valid
+        if (isset($transactionOptions['readOnly']) &&
+            is_array($transactionOptions['readOnly']) &&
+            isset($transactionOptions['readOnly']['readTime'])) {
+            $readTime = &$transactionOptions['readOnly']['readTime'];
+            $readTime = $this->parseCoreTimestamp($readTime);
```

Comment:
Avoid using pass-by-reference
```suggestion
            $readTime = $transactionOptions['readOnly']['readTime'];
            $transactionOptions['readOnly']['readTime'] = $this->parseCoreTimestamp($readTime);
```